nullaway/src/test/java/com/uber/nullaway/NullAwayTest.java

```diff
@@ -3170,4 +3170,31 @@ public void derefNullableTernary() {
             "}")
         .doTest();
   }
+
+  @Test
+  public void testCustomNullableAnnotation() {
+    makeTestHelperWithArgs(
+            Arrays.asList(
+                "-d",
+                temporaryFolder.getRoot().getAbsolutePath(),
+                "-XepOpt:NullAway:AnnotatedPackages=com.uber",
+                "-XepOpt:NullAway:CustomNullableAnnotation=qual.Null"))
+        .addSourceLines("qual/Null.java", "package qual;", "public @interface Null {", "}")
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import qual.Null;",
+            "class Test {",
+            "   @Null Object foo;", // No error, should detect @Null
+            "   @Null Object returnNullable(){",
+            "     passNullable(foo);",
+            "     return null;", // No error, should detect @Null
+            "   }",
+            "   String passNullable(@Null Object item){",
+            "    // BUG: Diagnostic contains: dereferenced expression",
+            "      return item.toString();",
+            "   }",
+            "}")
+        .doTest();
+  }
```

Comment:
This would need, at the very least, also a test case for the non-null case as well, note that for that you probably need a package in the unannotated set and the `AcknowledgeRestrictiveAnnotations` option.

Reply:
88f220fc1bf75db4ca58f354be29a11488ef00dd

To my understanding, ```NullAway``` only looks for  explicit ```@Nonnull``` annotation on method returns for checking inheritance conditions. This test case targets this property. Please let me know if I should add more cases. Thank you.

Reply:
Hey, I think this makes sense to me. As a sanity check, I'd try to comment the line:

```
"-XepOpt:NullAway:CustomNonnullAnnotations=qual.NoNull",
```

and make sure the test fails then.

That said, another thing I am going to ask you to do here is to separate this into two unit test methods, one for `testCustomNullableAnnotation()` and another for `testCustomNonnullAnnotation()`, just to keep each test case specific to one thing.

p.s. Given that separation, it might also be simpler in the `testCustomNonnullAnnotation` to just use a method of an unannotated class that takes `@NoNull` as an argument and pass `null` to it from annotated code, since that's easier to understand at a glance than the override/inheritance logic here.

Reply:
I tested it with the line ```"-XepOpt:NullAway:CustomNonnullAnnotations=qual.NoNull"``` commented and it did fail.

Also separated both tests with simpler logic: e2d4c4c7584ce1665d9592e4a75c2df75ca585a9